src/cat.js

```diff
@@ -37,6 +43,34 @@ function _cat(options, files) {
     cat += fs.readFileSync(file, 'utf8');
   });
 
+  if (options.number) {
+    cat = addNumbers(cat);
+  }
+
   return cat;
 }
 module.exports = _cat;
+
+function addNumbers(cat) {
+  var addEOF = false;
+  var lines = cat.split('\n');
+
+  if (lines[lines.length - 1] === '') {
+    addEOF = true;
+    lines.splice(-1);
+  }
+
+  lines = lines.map(function (line, i) {
+    return number(i + 1) + ' ' + line;
+  });
+
+  if (addEOF) {
+    lines.push('');
+  }
+
+  return lines.join('\n');
+}
+
+function number(n) {
+  return ('     ' + n).slice(-6);
```

Comment:
Add a comment? I think this does the right thing (prepends 5 spaces for 1-digit numbers, 4 spaces for 2-digit, 3 spaces for 3-digit, etc.), but it's not super obvious. A one-line comment explaining this would be great.

Reply:
Also, can we rename this function? "number" is vague. Maybe we can rename this to "prependPaddedNumber" and change its functionality to accept both a number and line (and it will insert all padding).